### DISCUSSION TITLE
Getting last 5 or 10 years Financial data in a dataframe?

### DISCUSSION THREAD

**Discussion by piyushdatta41** [dgunning/edgartools#164] [Q&A] [answered]
Hi guys! I am trying to find if there is a way to get the financial data of a company for the last five years in data frames?

**Comment by dgunning** [comment, marked as answer]
Use `MultiFinancials`

```python
>>> c = Company("AAPL")
>>> filings = c.latest("10-K", 4)
>>> financials = MultiFinancials(filings)
>>> financials.balance_sheet.to_dataframe()
```

**Reply by usamaalikhan** [reply]
Curious if there is a workaround the missing values that arise due to companies changing the account names? These missing values are not there when fetching individual filings and you can notice the account name difference.

**Comment by piyushdatta41** [comment]
Thanks for building this tool

Just curious to know if there any functions to get the detailed financial statements for in a dataframe? (for a range of years)

**Reply by otoburb** [reply]
The code that dgunning provided, especially the `to_dataframe()` method, will do what you want because it converts the financial statements from multiple years (in that example, 4 years) into a single dataframe you can then query.

**Reply by kalavrcom** [reply]
> Just curious to know if there any functions to get the detailed financial statements for in a dataframe? (for a range of years)

The best way to extract data from, and manipulate, Panadas dataframes can be found here: https://pandas.pydata.org/docs/
